### PR TITLE
Unique service names

### PR DESCRIPTION
**Pull request by shaynevanasperen**

### CODE REVIEW

File: test/Kongverge.Tests/Services/ConfigFileWriterTests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+{
+    [Story(Title = nameof(ConfigFileWriter))]
+    public class ConfigFileWriterTests : ScenarioFor<ConfigFileWriter>
+    {
+        private const string TestPath = nameof(TestPath);
+
+        public ConfigFileWriterTests() => Use<IFileProvider>(new VirtualFileProvider());
+
+        [BddfyFact(DisplayName = nameof(ConfigFileWriter.WriteConfiguration) + "NoServicesOrGlobalPlugins")]
+        public void Scenario1()
```
Comment (timvw74) [inline test/Kongverge.Tests/Services/ConfigFileWriterTests.cs:19]: Can we give these more descriptive names?

File: test/Kongverge.Tests/Services/ConfigFileReaderTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    [Story(Title = nameof(ConfigFileReader))]
+    public class ConfigFileReaderTests : ScenarioFor<ConfigFileReader>
+    {
+        private const string ExceptionThrownWithErrorMessages = "an exception is thrown showing error messages attributed to their respective file paths";
+        private const string TestPath = nameof(TestPath);
+
+        public ConfigFileReaderTests() => Use<IFileProvider>(new VirtualFileProvider());
+
+        [BddfyFact(DisplayName = nameof(ConfigFileReader.ReadConfiguration) + "MoreThanOneGlobalFile")]
+        public void Scenario1()
```
Comment (timvw74) [inline test/Kongverge.Tests/Services/ConfigFileReaderTests.cs:24]: Can we use descriptive names for the tests.

**Review by timvw74 — CHANGES_REQUESTED** [review]

File: test/Kongverge.Tests/Services/ConfigFileWriterTests.cs (hunk 1 above)
Comment (shaynevanasperen) [reply test/Kongverge.Tests/Services/ConfigFileWriterTests.cs:19]: I get you. Unfortunately this is a bit of a quirk with this testing framework, which is one factor that's beginning to make me wonder if I actually prefer the other style where you have a class per scenario and the tests are strung together based on conventional method names.

Regardless, this approach of naming the actual method without any descriptive name but then having an attribute above it that has the description, is a pattern that was already present in this code base. I haven't introduced this pattern as a new thing in this PR.

There appears to be some weird bug in `ReSharper`'s test runner that _sometimes_ results in the attributed `DisplayName` not being surfaced in the test runner window. You can even see in this screenshot that some tests show correctly while others don't:

image

However, I've found that `NCrunch` is better at this and _always_ shows the attributed `DisplayName` properly in the test runner window:

image

File: test/Kongverge.Tests/Services/ConfigFileReaderTests.cs (hunk 2 above)
Comment (shaynevanasperen) [reply test/Kongverge.Tests/Services/ConfigFileReaderTests.cs:24]: See answer above.

**Review by timvw74 — APPROVED** [review]

**Review by timvw74 — APPROVED** [review]

**Pull request merged**